src/repository/Repository.ts

```diff
@@ -38,10 +38,10 @@ export class Repository<ENTITY extends { _id?: ObjectId }> {
   }
 
   // mongodb returns null instead of undefined for optional fields
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  private mapNullToUndefined(entity: any): ENTITY {
+  private mapNullToUndefined(entity: WithId<ENTITY>): WithId<ENTITY> {
     const keys = Object.keys(entity);
-    const entries = keys.map((key) => [key, entity[key] ?? undefined]);
-    return Object.fromEntries(entries) as ENTITY;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const entries = keys.map((key) => [key, (entity as any)[key] ?? undefined]);
```

Comment:
Does it make sense to first claim entity to be a WithId<ENTITIY> but then cast it to any afterwards anyways? Hm I'm indecisive

Reply:
The function needs to return a `WithId<ENTITY>`, so I would like to at least let the type system check that we get a `WithId<ENTITY>` to begin with.
The cast is sadly needed to access via keys.